.github/workflows/therock_test_harness.yml

```diff
@@ -0,0 +1,98 @@
+name: TheRock Test Harness
+
+on:
+  workflow_dispatch:
+    inputs:
+      families:
+        type: string
+        description: 'The AMD GPU family to test. ex: gfx94X, gfx120X'
+        default: 'gfx94X'
+      release_version:
+        type: string
+        description: 'TheRock release version. (ex: nightly-tarball (X.Y.ZrcYYYYMMDD) or dev-tarball (X.Y.Z.dev0+{hash}))'
+        default: '7.9.0rc20251008'
+      tests_to_run:
+        type: string
+        description: 'The list of tests to run with "or" expression. (ex: "hipcub or rocprim")'
+        default: 'hipcub or rocprim or rocrand or rocthrust'
+
+permissions:
+  contents: read
+
+concurrency:
+  # A PR number if a pull request and otherwise the commit hash. This cancels
+  # queued and in-progress runs for the same PR (presubmit) or commit
+  # (postsubmit). The workflow name is prepended to avoid conflicts between
+  # different workflows.
+  group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  setup_metadata:
+    runs-on: ubuntu-24.04
+    outputs:
+      package_targets: ${{ steps.configure.outputs.package_targets }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          repository: "ROCm/TheRock"
+
+      - name: Setup Python
+        uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
+        with:
+          python-version: 3.12
+
+      - name: Generating package target matrix
+        id: configure
+        env:
+          AMDGPU_FAMILIES: ${{ inputs.families }}
+          THEROCK_PACKAGE_PLATFORM: "linux"
+          TEST_HARNESS_TARGET_FETCH: true
+        run: python ./build_tools/github_actions/fetch_package_targets.py
+
+
+  therock_test_harness_linux:
+    name: TheRock Tests Sharded Linux Nightly
```

Comment:
Let's iterate on this name too.
* Drop "TheRock"
* Maybe drop "Nightly" (especially if this has other triggers)
* Check that the name doesn't get too long in the github UI for this job and any child jobs (important information like which component is being tested, on which operating system, for which versions of different packages should be near the front of the job name and not elided at character limits)